db/view/view.cc

```diff
@@ -1301,16 +1303,19 @@ future<> mutate_MV(
                 return maybe_account_failure(std::move(f), std::move(*target_endpoint), false, updates_pushed_remote);
             });
             if (wait_for_all) {
-                fs->push_back(std::move(view_update));
+                fs.push_back(std::move(view_update));
             } else {
                 // The update is sent to background in order to preserve availability,
                 // its parallelism is limited by view_update_concurrency_semaphore
                 (void)view_update;
             }
         }
+        if (fs.size() >= max_concurrent_updates) {
+            co_await when_all_succeed(fs.begin(), fs.end());
```

Comment:
Like Avi said above, I think this is not the right way to limit concurrency. What you do is to split the work into batches of 128 items, and after starting each 128, you wait for all of them. This can lead to partial use of the resources while waiting for stragglers - if for example one out of 100 requests is always twice slower than the rest, half of the time you aren't doing any work. Another problem is that while you wait for the last item of work, the first ones have most likely long finished.
So the right way is to use a semaphore as I suggest in tutorial.md, or Avi said we even added a Seastar primitive to do it for you.